In [1]:
from Binaries import *

In [5]:
dead_time = 69

for index in [-1]:

    Buffer = RandomTrace("nuria", index)

    print(f"{index}/{Buffer.all_n_files} {Buffer.random_file}: {len(Buffer._these_traces)} traces" )

    station = np.array(Buffer._these_traces)
    I_vem = np.array([90. for i in range(3)])
    I_vem_high = I_vem + 5

    while True:

        pmt_has_converged = 0
        n_triggers = np.zeros(3)

        for pmt1, pmt2, pmt3 in zip(station[:,0], station[:,1], station[:,2]):

            stations = iter(zip(pmt1, pmt2, pmt3))

            for bins in stations:

                # assume all PMTs are working, k70 = 1.75
                # check the actual trigger rate
                if np.all(bins >= 1.75 * I_vem):
                    if np.any(bins >= 2.5 * I_vem):
                        
                        n_triggers += bins >= 2.5 * I_vem
                        break

        trigger_rates = n_triggers / (len(Buffer._these_traces) * 2048 * 8.33e-8)
        trigger_rates_high = n_triggers_high / (len(Buffer._these_traces) * 2048 * 8.33e-8)

        print(f"VEM_peak estimate: {I_vem} ===> {trigger_rates} Hz")
        print(f"VEM_peak estimate: {I_vem_high} ===> {trigger_rates_high} Hz")
        
        for i in range(3):
            if trigger_rates[i] <= 60: I_vem[i] -= 10
            elif 60 < trigger_rates[i] <= 68: I_vem[i] -= 1
            elif 68 < trigger_rates[i] <= 72: pmt_has_converged += 1
            elif 72 < trigger_rates[i] <= 80: I_vem[i] += 1
            elif 80 < trigger_rates[i]: I_vem[i] += 10

        break

        # if pmt_has_converged >= 3: break

-1/798 nuria_randoms0798.csv: 5000 traces
VEM_peak estimate: [90. 90. 90.] ===> [116.06204982 110.20033013 133.64720888] Hz
VEM_peak estimate: [95. 95. 95.] ===> [116.06204982 110.20033013 133.64720888] Hz


In [ ]:
n_Th_t2 = 0

Trigger = HardwareClassifier()

for n, (pmt1, pmt2, pmt3) in enumerate(zip(station[:,0], station[:,1], station[:,2])):

    trace = np.zeros((3,2048))

    trace[0] = np.array(pmt1) / I_vem[0]
    trace[1] = np.array(pmt2) / I_vem[1]
    trace[2] = np.array(pmt3) / I_vem[2]

    if Trigger.Th(3.2, trace):
        n_Th_t2 += 1

print(f"Th2 trigger rate: {n_Th_t2 / (2048 * 8.33e-9 * n)} Hz")



## old estimator

VEM_peak estimate: [120. 120. 120.] ===> [35.17031813 43.37672569 37.515006  ] Hz

VEM_peak estimate: [110. 110. 110.] ===> [63.30657263 69.16829232 62.13422869] Hz

VEM_peak estimate: [109. 110. 109.] ===> [65.6512605  69.16829232 63.30657263] Hz

VEM_peak estimate: [108. 110. 108.] ===> [67.99594838 71.51298019 65.6512605 ] Hz

VEM_peak estimate: [107. 110. 107.] ===> [72.68532413 71.51298019 69.16829232] Hz

VEM_peak estimate: [108. 110. 107.] ===> [69.16829232 71.51298019 69.16829232] Hz

Runtime: 6m 12.6s

In [ ]:
for _ in range(69):
    print(_, end = "\r")